src/parquet2/src/encoding/hybrid_rle/mod.rs

```diff
@@ -2,13 +2,47 @@
 mod bitmap;
 mod decoder;
 mod encoder;
+use std::fmt::Debug;
+
 pub use bitmap::{encode_bool as bitpacked_encode, BitmapIter};
 pub use decoder::Decoder;
 pub use encoder::{encode_bool, encode_u32};
 
 use crate::error::Error;
 
-use super::bitpacked;
+use super::bitpacked::{self, Unpackable};
+
+pub trait Decodable: Sized + Unpackable
+where
+    <Self as Unpackable>::Unpacked: Debug
+{
+    fn from_le_bytes(bytes: &[u8]) -> Self;
+    fn zero() -> Self;
+}
+
+impl Decodable for u32 {
+    fn from_le_bytes(pack: &[u8]) -> Self {
+        let mut bytes = [0u8; std::mem::size_of::<u32>()];
```

Comment:
why do you have to do this memcpy thing?
and not just do `u32::from_le_bytes(pack)`

Reply:
Ah turns out that this was because `from_le_bytes` expects a fixed size array instead of a slice of bytes.

We can do
```
            let bytes = pack.try_into().expect("Parquet should always encode u32 values");
            let value = u32::from_le_bytes(bytes);
```
which will do a runtime check that the slice is `[u8; 4]`, which would still be faster than copying the bytes over.

Reply:
As discussed offline, unfortunately in some cases `pack` contains less than 4 bytes (`test_invalid_utf8_parquet_reading` contains an example) so we can't do this.. reverting to the state prior that does byte-wise copying.